pkg/api-server/server.go

```diff
@@ -264,6 +264,12 @@ func addResourcesEndpoints(ws *restful.WebService, defs []model.ResourceTypeDesc
 	}
 }
 
+func ShouldBeReadOnly(kdsFlag model.KDSFlagType, cfg *kuma_cp.Config) bool {
+	return cfg.ApiServer.ReadOnly ||
+		(cfg.Mode == config_core.Global && kdsFlag.Has(model.FromZoneToGlobal) && !kdsFlag.Has(model.FromGlobalToZone)) ||
```

Comment:
`kdsFlag.Has(model.FromZoneToGlobal)` and `kdsFlag.Has(model.FromGlobalToZone)` are not going to be set for resources that are not synced at all (like RBAC). I think something more correct is:

```go
if cfg.ApiServer.ReadOnly {
   return true
}
if kdsFlag == model.KDSDisabled { // Add KDSDisabled = 0 to model
   return false 
}
if cfg.Mode == config_core.Global && !kdsFlag.Has(model.ProvidedByGlobal) {
  return true
}
if cfg.Mode == config_core.Zone && !kdsFlag.Has(model.ProvidedByZone) {
  return true
}
return false

Reply:
It looks better I found only one thing that looks strange
https://github.com/kumahq/kuma/blob/master/pkg/core/resources/apis/mesh/zz_generated.resources.go#L2806

Reply:
Yes these go both ways (From global to zone and from zone to global)